### PR TITLE
tweak lesson app

### PR DESCRIPTION
**Pull request by ann26**
This PR:
- [x] number the sections
- [x] match PDF name: <section_name>_<worksheet_name>
- [x] name the zip file like PDF file
- [x] add button to download PDF and zip 
- [x] hide item when the translation is not available
- [x] add button to download all relevant courses

Sections numbering
screenshot from 2019-02-12 18-05-50

Match title pdf
peek 2019-02-12 18-10

match title zip
peek 2019-02-12 18-12 b

This button will download the pdf module and sample test into one zip file
screenshot from 2019-02-12 18-07-06
peek 2019-02-12 18-13

When translation is not available, item is hidden
english
screenshot from 2019-02-12 18-08-19
indonesian
screenshot from 2019-02-12 18-08-24

### CODE REVIEW

File: django_project/lesson/views/worksheet.py [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
 
+import StringIO
+import os
+import zipfile
 from collections import OrderedDict
+from django.conf import settings
 from django.core.urlresolvers import reverse
-from django.http import HttpResponse
+from django.http import HttpResponse, JsonResponse
+from django.template import Template, Context
```
Comment (ismailsunni) [inline django_project/lesson/views/worksheet.py]: __[Flake8]__ [F401] 'django.template.Template' imported but unused

<sub>posted by Sider</sub>

File: django_project/lesson/views/worksheet.py [hunk 2]
```diff
@@ -306,3 +371,86 @@ def get_context_data(self, **kwargs):
...
+            zf.write(zip_data_path, zip_path)
+
+    zf.close()
+
+    zip_response = HttpResponse(
+        s.getvalue(), content_type="application/x-zip-compressed")
+    zip_response['Content-Disposition'] = \
+        'attachment; filename={}-worksheet.zip'.format(project.name.encode('utf8'))
+    return zip_response
+
```
Comment (ismailsunni) [inline django_project/lesson/views/worksheet.py]: __[Flake8]__ [W391] blank line at end of file

<sub>posted by Sider</sub>

File: django_project/lesson/views/worksheet.py [hunk 3]
```diff
@@ -306,3 +371,86 @@ def get_context_data(self, **kwargs):
...
+            zip_data_path = settings.MEDIA_ROOT + data_path[6:]
+            zip_path = os.path.join(zip_subdir, pdf_title + '.zip')
+            zf.write(zip_data_path, zip_path)
+
+    zf.close()
+
+    zip_response = HttpResponse(
+        s.getvalue(), content_type="application/x-zip-compressed")
+    zip_response['Content-Disposition'] = \
+        'attachment; filename={}-worksheet.zip'.format(project.name.encode('utf8'))
```
Comment (ismailsunni) [inline django_project/lesson/views/worksheet.py]: __[Flake8]__ [E501] line too long (83 > 79 characters)

<sub>posted by Sider</sub>

File: django_project/lesson/views/worksheet.py [hunk 4]
```diff
@@ -1,17 +1,23 @@
...
 """Worksheet views."""
 
+import StringIO
+import os
+import zipfile
 from collections import OrderedDict
+from django.conf import settings
 from django.core.urlresolvers import reverse
-from django.http import HttpResponse
+from django.http import HttpResponse, JsonResponse
```
Comment (ismailsunni) [inline django_project/lesson/views/worksheet.py]: __[Flake8]__ [F401] 'django.http.JsonResponse' imported but unused

<sub>posted by Sider</sub>

File: django_project/lesson/views/worksheet.py [hunk 5]
```diff
@@ -1,17 +1,23 @@
...
+from django.template import Template, Context
+from django.template.loader import get_template
```
Comment (ismailsunni) [inline django_project/lesson/views/worksheet.py]: __[Flake8]__ [F401] 'django.template.loader.get_template' imported but unused

<sub>posted by Sider</sub>

**Comment by ann26** [comment]
add button to download all relevant courses
peek 2019-02-13 11-02
peek 2019-02-13 11-13

**Comment by timlinux** [comment]
LGTM thanks @ann26! And thanks for the review @ismailsunni

**Comment by ismailsunni** [comment]
@timlinux haha, it's not me, but a bot from siderCI :D

**Pull request merged**